src/mscorlib/src/System/Runtime/CompilerServices/TaskAwaiter.cs

```diff
@@ -148,11 +155,11 @@ private static void HandleNonSuccessAndDebuggerNotification(Task task)
             task.NotifyDebuggerOfWaitCompletionIfNecessary();
 
             // And throw an exception if the task is faulted or canceled.
-            if (!task.IsCompletedSuccessfully) ThrowForNonSuccess(task);
+            return task.IsCompletedSuccessfully ? null : GetNonSuccessException(task);
         }
 
         /// <summary>Throws an exception to handle a task that completed in a state other than RanToCompletion.</summary>
```

Comment:
If this method changes to return rather than throw, the comment is wrong.

Reply:
Fixed